### PR TITLE
feat(snippet): add a snippet widget to be able to highlight snippet results

### PR DESCRIPTION
**Pull request by mthuret**
You can now write this: `<Snippet attributeName="description" hit={hit}/>` and have highlighting for snippet attributes.

#1776

### CODE REVIEW

File: packages/react-instantsearch/src/components/Highlight.js [hunk 1]
```diff
@@ -1,19 +1,6 @@
...
-  const reactHighlighted = parsedHighlightedValue.map((v, i) => {
-    const key = `split-${i}-${v.value}`;
-    if (!v.isHighlighted) {
-      return <span key={key} className="ais-Highlight__nonHighlighted">{v.value}</span>;
-    }
-    return <em key={key} className="ais-Highlight__highlighted">{v.value}</em>;
-  });
-  return <span className="ais-Highlight">{reactHighlighted}</span>;
+export default function Highlight(props) {
+  return <Highlighter path="_highlightResult" {...props}/>;
```
Comment (bobylito) [inline packages/react-instantsearch/src/components/Highlight.js]: Did you notice that `path` can be overriden?

File: packages/react-instantsearch/src/components/Highlight.js [hunk 2]
```diff
@@ -1,19 +1,6 @@
...
 }
-
-Highlight.propTypes = {
```
Comment (bobylito) [inline packages/react-instantsearch/src/components/Highlight.js]: How come we don't need to add the proptypes in this case?

File: packages/react-instantsearch/src/core/highlight.js [hunk 3]
```diff
@@ -5,25 +5,27 @@ import {get} from 'lodash';
...
   preTag = '<em>',
   postTag = '</em>',
+  path,
   attributeName,
   hit,
 }) {
   if (!hit) throw new Error('`hit`, the matching record, must be provided');
 
-  const highlightObject = get(hit._highlightResult, attributeName);
+  const highlightObject = get(hit[path], attributeName);
```
Comment (bobylito) [inline packages/react-instantsearch/src/core/highlight.js]: It's mostly a technicality but here `path` is not really a path but the property that contains the highlight structure. I suggest we change path to a name that explicitly state that the user can't go deep, idea: `highlightProperty`

File: packages/react-instantsearch/src/components/Highlight.js (hunk 1 above)
Comment (mthuret) [inline packages/react-instantsearch/src/components/Highlight.js]: Yes. The path is not a prop documented in the Highlight/Snippet widget so I figured out it was ok to just forward the props. Do you think it's an issue?

File: packages/react-instantsearch/src/components/Highlight.js (hunk 2 above)
Comment (mthuret) [inline packages/react-instantsearch/src/components/Highlight.js]: Well if I add them then you get two warnings if a props is missing. From the higlight, and from the highlighter. But I can definitely add them back.

File: packages/react-instantsearch/src/core/highlight.js (hunk 3 above)
Comment (mthuret) [inline packages/react-instantsearch/src/core/highlight.js]: Fine with me, I'll change for `highlightProperty`

File: packages/react-instantsearch/src/components/Highlight.js (hunk 1 above)
Comment (bobylito) [inline packages/react-instantsearch/src/components/Highlight.js]: Should be ok 👍

**Pull request merged**